storage/stores.go

```diff
@@ -23,25 +23,39 @@ import (
 	"golang.org/x/net/context"
 
 	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/gossip"
 	"github.com/cockroachdb/cockroach/keys"
 	"github.com/cockroachdb/cockroach/roachpb"
+	"github.com/cockroachdb/cockroach/storage/engine"
 	"github.com/cockroachdb/cockroach/util"
+	"github.com/cockroachdb/cockroach/util/hlc"
 	"github.com/cockroachdb/cockroach/util/log"
 	"github.com/cockroachdb/cockroach/util/tracer"
 )
 
-// A Stores provides methods to access a collection of local stores.
+// A Stores provides methods to access a collection of stores. There's
+// a visitor pattern and also an implementation of the client.Sender
+// interface which directs a call to the appropriate store based on
+// the call's key range. Stores also implements the gossip.Persistence
+// interface, which allows gossip bootstrap information to be
+// persisted consistently to every store and the most recent bootstrap
+// information to be read at node startup.
 type Stores struct {
-	mu       sync.RWMutex               // Protects storeMap and addrs
-	storeMap map[roachpb.StoreID]*Store // Map from StoreID to Store
+	clock      *hlc.Clock
+	mu         sync.RWMutex               // Protects storeMap and addrs
+	storeMap   map[roachpb.StoreID]*Store // Map from StoreID to Store
+	gossipBI   gossip.BootstrapInfo       // The latest gossip bootstrap info
```

Comment:
I'd like to avoid storing this in-memory copy since it isn't used for much. You could pass a value from `ReadBootstrapInfo` to `updateAllBootstrapInfos`, and in `AddStore` (which is rarely called) you could call `ReadBootstrapInfo` instead of using `updateAllBootstrapInfos`.